src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -1486,10 +1486,13 @@ export default class DefaultDeviceController
         };
         track.addEventListener('ended', newDevice.endedCallback, { once: true });
       }
-      newDevice.groupId = this.getGroupIdFromDeviceId(
-        kind,
-        this.getMediaTrackSettings(newDevice.stream)?.deviceId || ''
-      );
+      newDevice.groupId =
+        device !== null
+          ? this.getGroupIdFromDeviceId(
+              kind,
+              this.getMediaTrackSettings(newDevice.stream)?.deviceId || ''
+            )
+          : '';
```

Comment:
Just do `newDevice.groupId = '';` in the `null` section above, and then here you can do a simple conditional.

Reply:
The null section above is only for `audio` right?